### DISCUSSION TITLE
Is there a way to use preserveOrder selectively?

### DISCUSSION THREAD

**Discussion by adamalston** [NaturalIntelligence/fast-xml-parser#462] [Q&A]
I only want to use preserve order for a single XML element. Right now I am doing the following:

```js
preserveOrder: (name, jpath, isLeafNode, isAttribute) => {
  return 'description' === name || jpath.includes('description');
},
```

It's not working because `preserveOrder` is expecting a boolean:
```ts
type X2jOptions = {
  preserveOrder: boolean;
  ...
};
```

Is there a way to use preserveOrder selectively?

**Comment by amitguptagwl** [comment]
Currently, no. But ew can think about new feature.

**Reply by adamalston** [reply]
I think that would be a great feature to have.

**Reply by cgarrettjones** [reply]
is the preserveOrder option that operates on the entire parse the only method of ensuring order is kept? when the tool creates arrays or when isArray is used then are those array items in order irrespective of the preserveOrder option?

**Reply by amitguptagwl** [reply]
@cgarrettjones I hope you must have gone through the docs. When you use `preserveOrder` you get an ugly array structure. When you use `isArray` you get simplified array structure for elements you specified.

**Reply by cgarrettjones** [reply]
thanks @amitguptagwl. i have spatial points that represent a polygon in XML, its imperative that those point are in order in the outcome JSON array. does isArray preserve the order or must i use preserveOrder for the entire conversion? thanks for the detail!

**Comment by amitguptagwl** [comment]
@cgarrettjones better to use `preserveOrder`. `isArray` doesn't guarantee the order.

**Comment by amitguptagwl** [comment]
@cgarrettjones please check https://github.com/nodable/flexible-xml-parser once
